### DISCUSSION TITLE
[New item] Oil Reserve Canister

### DISCUSSION THREAD

**Discussion by TheDcol** [sunflower-land/sunflower-land#3833] [Ideas]
New Workbench placeable item.

```
Oil Reserve Caniste - 20 gold (or some other resource cost, this is an example)
Allows oil reserves to 'overfill' by x amount
```

Tired of waking up at weird times to harvest the oil? This Oil Reserve Canister allows your oil reserves to keep filling up over the max amount.

_EG. You harvest oil at 8am, you will then be able to harvest oil at 4am the next day (20 hours). If you wait until 8am the next day instead of 4am, you would have 'wasted' 4 hours that you could have collected oil for._

With the Oil Reserve Canister, oil keeps filling into its reserve!

_Using the same example above, you harvest oil at 8am, you will then be able to harvest oil at 4am the next day (20 hours).. With the Canister, if you wait until 8am the next day instead of 4am, you get the usual oil amount BUT oil would already have been collecting since 4am in your canister! Meaning, you did not miss out on the oil while you slept_

OPTIONS:
- Size of canister to be determined, calculated by hours or size?
- Different types of canisters in the workbench. Small (12.5), Medium (15), Large (20)
- Increase/decrease costs. 20 gold was a figure not thought about much.

**Comment by Dionis404** [comment]
Maybe make an option to automatically collect the oil when you put a drill on it? 

You buy a drill, put it on the oil, and when it's ready, the drill disappears, the oil is collected, and the countdown to a new collection starts and at that point you can already put a new drill on it. 

Or you can add a new build for each oil. 
Which allows you to install a drill for automatic collection.  

Something like a tower in which you need to install a drill, but it already collects the resource itself.

**Reply by TheDcol** [reply]
> Something like a tower in which you need to install a drill, but it already collects the resource itself.

Hmm interesting...

- Buy 'oil tower'
- Place tower on top of oil well
- Max 10 drills can be added to tower at once.
- Tower will automatically harvest the oil 10 times once filled

Something like that?

**Reply by Dionis404** [reply]
Something like that.  
I think this system will bring new mechanics to resource extraction in the game.  
I wonder what Adam and others think about this. 

We can come up with something similar for all kinds of resource that have a bonus on N -day 
Perhaps FarmHand could be involved for such automation as well.

**Comment by brian-man** [comment]
Thanks for this discussion!

Something to consider:
Generalize the concept of "overfill" buffer/reservoir.  Initially support oil.  In the future (for a cost), you could add Honey (for beehives), Crimstone, Sunstone, etc.  The generalized concept is "allow the thing to continue producing, but still require manual tapping to collect" as a countermeasure to both drift and sleep deprivation.

Dionis's concept of queuing or automating is also nice to consider, but kind of feels like a v2 or v3 evolution of the original concept.

**Reply by Dionis404** [reply]
Here is where I wrote about the idea in more detail
https://github.com/sunflower-land/sunflower-land/discussions/3855
